tests/cli/test_main_clean.py

```diff
@@ -0,0 +1,235 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2012 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+
+from glob import glob
+from json import loads as json_loads
+from os import listdir
+from os.path import basename, isdir, join, exists
+from shutil import copy
+import pytest
+from conda.base.constants import CONDA_TEMP_EXTENSION
+from conda.core.subdir_data import create_cache_dir, SubdirData
+from conda.testing.integration import make_temp_package_cache, run_command, Commands, make_temp_env
+
+
+def _get_contents(pkgs_dir):
+    return [join(pkgs_dir, d) for d in listdir(pkgs_dir)]
+
+
+def _get_pkgs(pkgs_dir=None, *, contents=None):
+    return [d for d in contents or _get_contents(pkgs_dir) if isdir(d)]
+
+
+def _get_tars(pkgs_dir=None, *, contents=None):
+    return [t for t in contents or _get_contents(pkgs_dir) if t.endswith((".tar.bz2", ".conda"))]
+
+
+def _get_index_cache():
+    return glob(join(create_cache_dir(), "*.json"))
+
+
+def _get_tempfiles(pkgs_dir=None, *, contents=None):
+    return [
+        t
+        for t in contents or _get_contents(pkgs_dir)
+        if t.endswith((".trash", CONDA_TEMP_EXTENSION))
+    ]
+
+
+def _get_all(pkgs_dir):
+    contents = _get_contents(pkgs_dir)
+    return _get_pkgs(contents=contents), _get_tars(contents=contents), _get_index_cache()
+
+
+def _any_pkg(name, contents):
```

Comment:
Since this function is used only for checking assertions I'd say it'd be fine to rename it to `assert_any_pkg` and do the assert check inside, would cleanup the tests below a bit more.

```
assert _any_pkg(pkg, _get_pkgs(pkgs_dir))
```
vs.

```
assert_any_pkg(pkg, pkgs_dir)
```

Reply:
What do we do with the `assert not _any_pkg()` cases?

Reply:
ah, yeah, the pattern used in pytest heavy codebases is implementing `assert_not_any_pkg` or `assert_no_pkg`?